### ISSUE TITLE
Wrong spatial subset in make_local

### ISSUE THREAD

**Issue by JanisGailis** [CCI-Tools/cate-core#196]
Doing
```python
zemgale = '23.5, 56.3, 25.0, 57.0'
ds = soil_remote_sources[0].make_local(local_name='sm.data',
                                       time_range='2000-01-01, 2010-12-30',
                                       region=zemgale,
                                       var_names='sm, sm_uncertainty',
                                       monitor=monitor)
```

returns a dataset with following spatial dimensions:
```python
ds['lon']
```
```shell
<xarray.DataArray 'lon' (lon: 3)>
array([ 56.375,  56.625,  56.875], dtype=float32)
Coordinates:
  * lon      (lon) float32 56.375 56.625 56.875
```

```python
ds['lat']
```
```shell
<xarray.DataArray 'lat' (lat: 6)>
array([-23.625, -23.875, -24.125, -24.375, -24.625, -24.875], dtype=float32)
Coordinates:
  * lat      (lat) float32 -23.625 -23.875 -24.125 -24.375 -24.625 -24.875
```

First of all, the sign is wrong. But also lat and lon have been mixed. E.g. doing
```python
reg = PolygonLike.convert(zemgale)
reg.bounds
```
yields:
```shell
(23.5, 56.3, 25.0, 57.0)
```
where the result means:
(x_min, y_min, x_max, y_max), or in our lon/lat datasets: (lon_min, lat_min, lon_max, lat_max)

**Comment by JanisGailis** [comment]
@kbernat Have you taken notice of this?

**Comment by kbernat** [comment]
@JanisGailis yep, will take care of it today

**Comment by kbernat** [comment]
@JanisGailis 
I fixed part of the issue - mixed lat/lon params order but the negative value is still a problem.
Looks like data in this ECV is stored quite different way. I checked with ncdump and lat/lon values looks like this:
```
data:

 lon = -179.875, -179.625, -179.375, -179.125, -178.875, -178.625, -178.375, 
       ...
       -0.375, -0.125, 0.125, 0.375, 0.625, 0.875, 1.125, 1.375, 1.625, 1.875, 
       ...
       177.625, 177.875, 178.125, 178.375, 178.625, 178.875, 179.125, 179.375, 
       179.625, 179.875 ;

 lat = 89.875, 89.625, 89.375, 89.125, 88.875, 88.625, 88.375, 88.125, 
       ...
       1.625, 1.375, 1.125, 0.875, 0.625, 0.375, 0.125, -0.125, -0.375, -0.625, 
       ...
       -87.875, -88.125, -88.375, -88.625, -88.875, -89.125, -89.375, -89.625, 
       -89.875 ;

```
The problem is that other ECVs (e.g. SST) use different order (-90...90) but the same units (degree_north)

**Comment by kbernat** [comment]
None of local attrs (valid_range, unit) or global attrs (geospatial_lat_min, geospatial_lat_max, geospatial_lat_units) are helpful in this case. 
I use isel (index selection) to avoid downloading unnecessary data and looks like I can rely only on global attributes which may contain some buggy data.
Any ideas how to process that?

**Comment by JanisGailis** [comment]
What ```isel``` is that? As in xr.isel or some openDAP thing?

**Comment by JanisGailis** [comment]
I see two possible paths forward.

1. If that's xarray isel you're talking about, use ```sel```, or something equivalent if it's not, that works on data not indexes.

2. Retrieve the full lon/lat and do some automatic introspection on how the data is laid out to figure out which indexes to use for selection.

But yes, this sucks.

**Comment by kbernat** [comment]
I will ask ODP team why it works like this, for now we need workaround anyway and I vote for second option

1. xr.sel requires to load whole data up front
2. Yes, that may be only "easy" solution. It will give +3 http requests to thredds server but still more efficient than letting xr.sel to download whole data just to check indexes internally.

**Comment by mzuehlke** [comment]
Xarray.sel requires that the values of the slices are increasing or decreasing as the values in the coordinate are.

**Comment by kbernat** [comment]
That's true but for now I cant jump over "The indexing operation you are attempting to perform is not valid on netCDF4.Variable object. Try loading your data into memory first by calling .load()."
And if I use simply load() it will fetch whole data.

**Comment by kbernat** [comment]
@JanisGailis subsetting works fine now, but I didnt change the data order.

**Comment by JanisGailis** [comment]
Thanks! I'll test shortly!

**Comment by JanisGailis** [comment]
Fixed with #200
